src/test/kotlin/com/example/api/controller/BookingControllerIntegrationTest.kt

```diff
@@ -0,0 +1,94 @@
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest
+import org.springframework.boot.test.mock.mockito.MockBean
+import org.springframework.test.context.ContextConfiguration
+import org.springframework.test.context.junit.jupiter.SpringExtension
+import org.springframework.test.web.servlet.MockMvc
+import org.springframework.http.MediaType
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers
+
+@ExtendWith(SpringExtension::class)
+@WebMvcTest(controllers = [BookingController::class])
+@ContextConfiguration
+class BookingControllerIntegrationTest {
+    @Autowired
+    private lateinit var mockMvc: MockMvc
+
+    @Autowired
+    private lateinit var objectMapper: ObjectMapper
+
+    @MockkBean
+    private lateinit var bookingService: BookingService
+
+    @Test
+    internal fun `Should create booking`(){
+        every { bookingService.addBooking(1, fakeBookingRequest) }
+
+        mockMvc.perform(
+            MockMvcRequestBuilders
+                .post("/customer/${fakeCustomer.customerId}/booking")
+                .content(objectMapper.writeValueAsBytes(fakeBookingRequest))
+                .contentType(MediaType.APPLICATION_JSON)
+        ).andExpect(MockMvcResultMatchers.status().isCreated)
+            .andExpect(
+                MockMvcResultMatchers.content().json(objectMapper.writeValueAsString(fakeBookingRequest.copy(id = 1)))
+            )
+    }
+
+    @Test
+    internal fun `Should get the booking of a customer`() {
+        every { bookingService.getBooking(1,1) } returns fakeBooking
+
+        mockMvc.perform(
+            MockMvcRequestBuilders
+                .get("/customer/${fakeCustomer.customerId}/booking/${fakeBooking.id}")
+                .contentType(MediaType.APPLICATION_JSON)
+        )
+            .andExpect(MockMvcResultMatchers.status().isOk)
+            .andExpect(MockMvcResultMatchers.content().json(objectMapper.writeValueAsString(fakeBookingRequest)))
+    }
+
+    @Test
+    internal fun `Should delete booking`() {
+        every { bookingService.deleteBooking(1,1) }
+
+        mockMvc.perform(
+            MockMvcRequestBuilders
+                .delete("/customer/${fakeCustomer.customerId}/booking/${fakeBooking.id}")
+                .contentType(MediaType.APPLICATION_JSON)
+        )
+            .andExpect(MockMvcResultMatchers.status().isNoContent)
+    }
+
+    @Test
+    fun `Should update a booking`(){
+        every { bookingService.updateBooking(1,1,fakeBookingRequest.copy(Name = "Jess")) }
+
+        mockMvc.perform(
+            MockMvcRequestBuilders
+                .put("/customer/${fakeCustomer.customerId}/booking/${fakeBooking.id}")
+                .content(objectMapper.writeValueAsBytes(fakeBookingRequest))
+                .contentType(MediaType.APPLICATION_JSON)
+        ).andExpect(MockMvcResultMatchers.status().isNoContent)
```

Comment:
Here also it's ResponeEntity.ok in your controller and you used isNoContent here?